mimic/model/cloudfeeds.py

```diff
@@ -0,0 +1,86 @@
+"""
+Emulate the behavior of Rackspace Cloud Feeds service.
+"""
+
+import attr
+
+
+@attr.s
+class CloudFeedsProduct(object):
+    """
+    Models a single CloudFeed product endpoint and its respective
+    functionality.
+    """
+    title = attr.ib(attr.validators.instance_of(str))
+    href = attr.ib(attr.validators.instance_of(str))
+
+    events = attr.ib(default=attr.Factory(list))
+
+    def post(self, event):
+        """
+        Post an event to the product queue.
+        """
+        self.events.append(event)
+
+
+@attr.s
+class CloudFeeds(object):
+    """
+    Models CloudFeeds support at the plugin-level.
+    """
+    _endpoints = attr.ib(default=attr.Factory(dict))
+
+    def get_product_endpoints(self):
+        """
+        Return a list of product endpoints registered with this class.
+        """
+        return dict(self._endpoints)
+
+    def register_product(self, title, href):
+        """
+        Register a product by creating a URL that one can GET and/or POST from/to,
+        and providing a descriptive title for it in the product endpoint
+        listing.
```

Comment:
Non-blocking doc nitpick: since there is an explicit test to ensure that this just does nothing if you attempt to re-register with either the same or a different title, maybe we should document that?